lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_template.py

```diff
@@ -286,7 +295,11 @@ def main():
         state=dict(choices=['present', 'absent'], default='present'),
     )
 
-    module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
+    module = TowerModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[('credential', 'credentials')]
```

Comment:
You should also make it exclusive with `vault_credential`, which would be another tuple in this list.